src/operations/execute_operation.ts

```diff
@@ -82,11 +82,6 @@ export async function executeOperation<
   } else if (session.client !== client) {
     throw new MongoInvalidArgumentError('ClientSession must be from the same MongoClient');
   }
```

Comment:
Good catch, I did intentionally remove this because the options are combined at this point that I don't know if timeoutMS is set because it was placed on the collection/db/client vs passed directly to the operation. Added tests and the validation lives in resolve options.